### PR TITLE
a11y: impress: screen reader support for text shape editing

### PR DESCRIPTION
**Pull request by mcecchetti**
Now accessibility support can be enabled in Impress.
Created cypress tests for editable area in Impress

Signed-off-by: Marco Cecchetti <marco.cecchetti@collabora.com>
Change-Id: Ia2fd4e55bce3785320ec0cc9f31a6d7550ca3a82

### CODE REVIEW

**Comment by mcecchetti** [comment]
For passing cypress tests it need core patch: https://gerrit.libreoffice.org/c/core/+/157525 to be merged

File: browser/src/layer/marker/TextInput.js [hunk 1]
```diff
@@ -1244,14 +1248,19 @@ L.TextInput = L.Layer.extend({
...
+					if (!this._map._docLayer || !this._map._docLayer._visibleCursor)
+						return;
+					window.app.console.log('A11yTextInput._setAcceptInput: going to emit a synthetic click after switching to edit mode.');
+					var top = this._map._docLayer._visibleCursor.getNorthWest();
+					var bottom = this._map._docLayer._visibleCursor.getSouthWest();
+					var center = L.latLng((top.lat + bottom.lat) / 2, top.lng);
+					var cursorPos = this._map._docLayer._latLngToTwips(center);
+					this._map._docLayer._postMouseEvent('buttondown', cursorPos.x, cursorPos.y, 1, 1, 0);
+					this._map._docLayer._postMouseEvent('buttonup', cursorPos.x, cursorPos.y, 1, 1, 0);
+				}.bind(this), 100);
```
Comment (eszkadev) [inline browser/src/layer/marker/TextInput.js:1263]: can we do that more deterministic?
that after required event we call this code

Comment (mcecchetti) [reply browser/src/layer/marker/TextInput.js]: I pushed a new version that doesn't need delay. I restricted the code to text document only.

**Review by eszkadev — APPROVED** [review]
now it looks ok

**Comment by eszkadev** [comment]
cypress was failing on first attempt on:
✖  impress/editable_area_spec.js            04:20        4        1        3

**Comment by mcecchetti** [comment]
It seems that core patch has not yet available in this build, hence impress/editable_area.js is failing; 
anyway since core patch has already been merged I think it's better to merge this one soon in order to have it in CODE 23.05.5.1

**Pull request merged**